psi4/src/psi4/cc/amps.cc

```diff
@@ -32,35 +32,39 @@
 #include "psi4/libmints/dimension.h"
 #include "psi4/libmints/matrix.h"
 #include "psi4/libpsi4util/process.h"
+#include "psi4/libmints/molecule.h"
 #include <memory>
 namespace psi {
 namespace ccenergy {
+
 std::map<std::string, SharedMatrix> CCEnergyWavefunction::get_amplitudes() {
     // re-init dpd
     int** cachelist;
     auto cachefiles = std::vector<int>(PSIO_MAXUNIT);
+
     auto ref = options_.get_str("REFERENCE");
-    if (ref == "ROHF" && options_.get_bool("SEMICANONICAL")){
-      ref = "UHF";
-    }
+    if (ref == "ROHF" && options_.get_bool("SEMICANONICAL")) ref = "UHF";
+
+    const std::string pg = molecule_->sym_label();
+    if (nirrep_ > 1) throw InputException("Point-group symmetry must be c1", "symmetry", pg, __FILE__, __LINE__);
+
     std::vector<DPDMOSpace> spaces;
-    std::map<std::string, SharedMatrix> amps;
     if (dpd_list[0] == nullptr) {
         if (ref == "RHF") {
-            Dimension occpi_ = nalphapi_;
+            Dimension occpi_ = nalphapi_ - frzcpi_;
             Dimension virtpi_ = nmopi_ - nalphapi_;
             cachelist = cacheprep_rhf(options_.get_int("CACHELEVEL"), cachefiles.data());
             spaces = {DPDMOSpace{'o', "ijkl", occpi_}, DPDMOSpace{'v', "abcd", virtpi_}};
         } else if (ref == "ROHF") {
-            Dimension doccpi_ = nalphapi_;
-            Dimension soccpi_ = nbetapi_ - nalphapi_;
+            Dimension doccpi_ = nbetapi_ - frzcpi_;
+            Dimension soccpi_ = nalphapi_ - nbetapi_;
```

Comment:
```suggestion
            Dimension soccpi_ = nalphapi_ - nbetapi_ - frzcpi_;
```

Otherwise `soccpi_` contains the frozen core electrons, which is wrong.

Reply:
I believe this would contain frozen electrons only if the multiplicity is large enough. `nalpha_` and `nbeta_` both count for the frozen electrons, so they cancel out.

Reply:
Maybe an if statement to check that `soccpi_` is not larger than `nalphapi_ - `frzcpi_`?

Reply:
Ah, nevermind.

Does Psi4 enforce `M >= 1`? If not, `nbetapi_` can be larger than `nalphapi_`

Reply:
I'm not completely sure about that. I assumed that was the case from looking at the HF code, but maybe not. Will ask around.

Reply:
Justin confirms that all unpaired electrons are of alpha spin.